lib/hide-sensitive.js

```diff
@@ -3,7 +3,8 @@ const {SECRET_REPLACEMENT, SECRET_MIN_SIZE} = require('./definitions/constants')
 
 module.exports = (env) => {
   const toReplace = Object.keys(env).filter(
-    (envVar) => /token|password|credential|secret|private/i.test(envVar) && size(env[envVar].trim()) >= SECRET_MIN_SIZE
+    (envVar) =>
+      /token|password|credential|secret|(?<!^go)private/i.test(envVar) && size(env[envVar].trim()) >= SECRET_MIN_SIZE
```

Comment:
I'm not 100% sure about adding a workaround for one specific environment variable, but I have no better idea either. What I would do however is to make the code more readable and to comment the workaround. I'd suggest to do the following

```suggestion
    (envVar) => {
      // https://github.com/semantic-release/semantic-release/issues/1558
      if (envVar === 'GOPRIVATE') {
        return false
      }
      
      return /token|password|credential|secret|private/i.test(envVar) && size(env[envVar].trim()) >= SECRET_MIN_SIZE
    }
```

Reply:
Sounds good @gr2m.  I thought about adding a debug line to print out matched env vars... that sure would have helped me figure this one out a lot sooner.